Description: drop WWW::Mechanize::DecodedContent, WWW::Mechanize's content() decodes since 1.50
Origin: vendor
Bug-Debian: http://bugs.debian.org/636690
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>

--- a/META.yml
+++ b/META.yml
@@ -18,7 +18,6 @@
 requires: 
   HTML::AutoPagerize: 0
   JSON: 2.09
-  WWW::Mechanize: 0
-  WWW::Mechanize::DecodedContent: 0
+  WWW::Mechanize: 1.50
   perl: 5.8.1
 version: 0.02
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -3,8 +3,7 @@
 all_from 'lib/WWW/Mechanize/AutoPager.pm';
 
 requires 'HTML::AutoPagerize';
-requires 'WWW::Mechanize';
-requires 'WWW::Mechanize::DecodedContent';
+requires 'WWW::Mechanize' => '1.50';
 requires 'JSON' => 2.09;
 
 build_requires 'Test::More';
--- a/lib/WWW/Mechanize/AutoPager.pm
+++ b/lib/WWW/Mechanize/AutoPager.pm
@@ -6,7 +6,7 @@
 
 use HTML::AutoPagerize;
 use Scalar::Util qw( weaken );
-use WWW::Mechanize::DecodedContent;
+use WWW::Mechanize;
 use JSON;
 
 sub WWW::Mechanize::autopager {
@@ -72,7 +72,7 @@
 sub next_link {
     my $self = shift;
 
-    my $res = $self->{autopager}->handle($self->{mech}->uri, $self->{mech}->decoded_content)
+    my $res = $self->{autopager}->handle($self->{mech}->uri, $self->{mech}->content)
         or return;
 
     return $res->{next_link};
@@ -81,7 +81,7 @@
 sub page_element {
     my $self = shift;
 
-    my $res = $self->{autopager}->handle($self->{mech}->uri, $self->{mech}->decoded_content)
+    my $res = $self->{autopager}->handle($self->{mech}->uri, $self->{mech}->content)
         or return;
 
     return $res->{page_element};
